include/deal.II/fe/fe_p1nc.h

```diff
@@ -0,0 +1,311 @@
+ *
+ *  @verbatim
+ *  2---------|---------3
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  -                   -
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  0---------|---------1
+ *  @endverbatim
+ *
+ * For each vertex $v_j$ of given quad, there are two edges of which $v_j$ is one of end points.
+ * Consider a linear function such that 0.5 value at two midpoints of such edges,
+ * and 0.0 at two midpoints of other edges.
+ * Note that the set of these values satisfies the dice rule which is described above.
+ * We denote such a function associated with vertex $v_j$ by $\phi_j$.
+ *
+ * The following figures represent $\phi_j$, $j=0,\cdots,3$ with its values at midpoints.
+ * Canonical (local) basis functions are given by any three shape functions among
+ * the following four linear functions.
+ *
+ * <ul>
+ * <li> shape function $\phi_0$:
+ *  @verbatim
+ *  +--------0.0--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.5                 0.0
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.5--------+
+ *  @endverbatim
+ *
+ * <li> shape function $\phi_1$:
+ *  @verbatim
+ *  +--------0.0--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.0                 0.5
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.5--------+
+ *  @endverbatim
+ *
+ * <li> shape function $\phi_2$:
+ *  @verbatim
+ *  +--------0.5--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.5                 0.0
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.0--------+
+ *  @endverbatim
+ *
+ * <li> shape function $\phi_3$:
+ *  @verbatim
+ *  +--------0.5--------+
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ * 0.0                 0.5
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  +--------0.0--------+
+ *  @endverbatim
+ *
+ * </ul>
+ *
+ * Note that above shape functions are constructed on each cell, not on the reference cell only.
+ * @p get_linear_shape computes the coefficients for shape functions when @p fill_fe_values
+ * is called on each cell.
+ *
+ * The basis function associated with a node in global scope is defined by the composition of
+ * (local) basis functions associated with the node on each element.
+ * And DOF associated with the node represents the coefficient of the basis function associated with the node.
+ * In this frame, all DOFs are truly independent.
```

Comment:
Hm, this is an interesting statement. Are you saying that (i) restricted to a single cell, these four shape functions are linearly dependent, (ii) but if you consider these shape functions globally, one per vertex, then they are linearly independent? Is this true for any kind of mesh? What if you had a mesh with only one cell, for example?